### DISCUSSION TITLE
Make edge data NOT optional? (v10)

### DISCUSSION THREAD

**Discussion by Sec-ant** [xyflow/xyflow#1923] [Ideas]
A related propose of this issue: https://github.com/wbkd/react-flow/issues/1679 with similar reasons.

There're custom nodes and custom edges, and `data` is kind of the only way to pass custom data from RF to your custom components, or you'll have to use contexts or stores. Now that `data` is required in nodes in v10, will `data` in edges be required, too? Also, different optionality of `data` type in nodes and edges is a little inconsistent.

**Comment by aamalric-talend** [comment]
Hello,
This could be great to move edge `data` property from optional to required.
